crates/weaver_resolved_schema/src/v2/event.rs

```diff
@@ -0,0 +1,69 @@
+//! Event related definition structs.
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use weaver_semconv::{
+    attribute::RequirementLevel,
+    v2::{signal_id::SignalId, CommonFields},
+};
+
+use crate::v2::attribute::AttributeRef;
+
+/// The definition of an Event signal.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct Event {
+    /// The name of the event.
+    pub name: SignalId,
+
+    /// List of attributes that belong to this event.
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub attributes: Vec<EventAttributeRef>,
+
+    // TODO - Should Entity Associations be "strong" links?
+    /// Which resources this span should be associated with.
+    ///
+    /// This list is an "any of" list, where a metric may be associated with one or more entities, but should
+    /// be associated with at least one in this list.
```

Comment:
I thought about entity associations as all-of rather than any-of - perhaps we should document whatever it is somewhere in the semconv repo and come back here to update docs?